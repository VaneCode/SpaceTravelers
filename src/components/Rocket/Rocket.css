.flag {
  letter-spacing: -0.18px;
  padding: 0.02rem 0.5rem;
  border-radius: 0.2rem;
  margin-right: 0.5rem;
  color: white;
  background-color: #19a2b8;
  font-weight: bold;
  font-size: 0.7rem;
}
