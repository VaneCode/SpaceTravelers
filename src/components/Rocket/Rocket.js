import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Card } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { HandledBookRocket } from '../../redux/rockets/rockets';

const Rocket = (props) => {
  const dispatch = useDispatch();
  const { rocket } = props;
  const ReserveHandler = () => {
    dispatch(HandledBookRocket({ rocket }));
  };

  return (
    <div className="row no-gutters">
      <div className="col-auto">
        <Card.Img
          className="img-fluid"
          style={{ width: '15rem', height: '10rem' }}
          src={rocket.flickr_images}
          alt={rocket.rocket_name}
        />
      </div>
      <div className="col">
        <Card.Body>
          <Card.Title className="h2" style={{ marginBottom: '1rem' }}>
            {rocket.rocket_name}
          </Card.Title>
          <Card.Text style={{ textAlign: 'justify' }}>
            {rocket.description}
          </Card.Text>
          <Button
            onClick={ReserveHandler}
            variant={rocket.reserved ? 'outline-secondary' : 'primary'}
          >
            {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rockect'}
          </Button>
        </Card.Body>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.number,
    rocket_name: PropTypes.string,
    description: PropTypes.string,
    flickr_images: PropTypes.instanceOf(Array),
    reserved: PropTypes.bool,
  }).isRequired,
};

export default Rocket;
