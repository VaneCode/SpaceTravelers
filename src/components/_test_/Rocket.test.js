import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import store from "../../redux/configureStore";
import Rocket from "../Rocket/Rocket";

describe("Rocket test", () => {
  test("snapshot", () => {
    const rocket = {
      id: 1,
      rocket_name: "Falcon1",
      description: "Fake description",
      flickr_images:
        "https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg",
      reserved: false,
    };
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Rocket rocket={rocket}/>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
