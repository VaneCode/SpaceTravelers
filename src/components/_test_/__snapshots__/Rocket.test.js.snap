// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket test snapshot 1`] = `
<div
  className="row no-gutters"
>
  <div
    className="col-auto"
  >
    <img
      alt="Falcon1"
      className="card-img img-fluid"
      src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
      style={
        Object {
          "height": "10rem",
          "width": "15rem",
        }
      }
    />
  </div>
  <div
    className="col"
  >
    <div
      className="card-body"
    >
      <div
        className="h2 card-title h5"
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        Falcon1
      </div>
      <p
        className="card-text"
        style={
          Object {
            "textAlign": "justify",
          }
        }
      >
        Fake description
      </p>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Reserve Rockect
      </button>
    </div>
  </div>
</div>
`;
